app/domain/authentication/authn_oidc/authenticator.rb

```diff
@@ -26,6 +23,9 @@ def call
       def validate_id_token_claims
         expected = { client_id: client_id, issuer: issuer } # , nonce: 'nonce'}
         @user_details.id_token.verify!(expected)
+      rescue => e
+        raise OIDCAuthenticationError, e.message if is_invalid_token_error?(e)
+        raise e
       end
```

Comment:
Rather than rescue all (which you should never do), check for something specific, and then re-raise for all others, simply do: 

```
      def validate_id_token_claims
        expected = { client_id: client_id, issuer: issuer } # , nonce: 'nonce'}
        @user_details.id_token.verify!(expected)
      rescue OpenIDConnect::ResponseObject::IdToken::InvalidToken => e
        raise OIDCAuthenticationError, e.message
      end
```
This will remove the need for the `is_invalid_token_error?` method.

As a side note for future reference, when naming a method that returns a boolean, use the `?` suffix but not the `is_` prefix, since using both is redundant.  Ie, the above method would be named `invalid_token_error?`.  Otoh, variable names cannot end with a `?`, so in that case you can use the `is_` prefix where appropriate.